### ISSUE TITLE
What is the rule to save the best model?

### ISSUE THREAD

**Issue by GOZGOXGOYGO** [MIC-DKFZ/nnUNet#408]
Hi Fabian,

thanks a lot for your fantastic work. However, I am not specific about the rule to save the best model. It seems that there is a criterion related to the dice on validation set, and the criterion will iterate with C(n+1) = 0.9 * C(n) + 0.1 * New dice. The best model will be saved when criterion surpass the best.

I am not sure if it's right? I take out the `all_val_eval_metrics` and calculate according to this rule, but the result seems inconsistent with what nnU-Net did.

Best,
Frank

**Comment by FabianIsensee** [comment]
Hi Frank,
nnU-Net uses all_val_eval_metrics for that. I just looked at the code and it seems good to me. If you can provide a concrete example where this fails then I can look into it further.
Best,
Fabian
PS: I do not use the best model ever. It does not outperform the final model most of the time

**Comment by GOZGOXGOYGO** [comment]
@FabianIsensee 
However, if I train the model with  argument ['all'], how to garantee that the final model will not overfit？By the way, may I ask what is the difference between the best model and the final model? What are the advantages of the final model? Thanks a lot!

Best,
Frank

**Comment by FabianIsensee** [comment]
Hi Frank,
ignore the model_best. It is a remnant of a long time ago. nnU-Net always uses the final model. nnU-Net also completely ignores overfitting, even in the cross-validation so there is no need to control it when training with all
